api/apps/api/src/modules/users/users.controller.ts

```diff
@@ -198,4 +199,21 @@ export class UsersController {
       throw new ForbiddenException();
     }
   }
+
+  @ApiOperation({
+    description: 'Find users by full email address.',
+  })
+  @ApiOkResponse()
+  @ApiUnauthorizedResponse()
+  @ApiForbiddenResponse()
+  @Get('by-email')
+  async findByEmail(
+    @Body() dto: UpdateUserDTO,
```

Comment:
I would suggest to accept the email address as a `@Query()` (`?email=abc@example.com`) or `@Param()` (`/api/v1/users/by-email/abc@example.com`) - we shouldn't rely on a body in a GET request.

Also, and following from that, we would accept a plain string (in terms of TS typing), not a subset of a DTO such as UpdateUserDTO (in general, I'd recommend against repurposing a DTO like this, in any case - best to create a small ad-hoc DTO if really needed, or using tiny-types for expressivity).

Reply:
I would also add an `@IsEmail()` validator (this is also missing in my original `CreateUserDTO`, although it is present in `LoginDto` at least 🤦🏼)